### ISSUE TITLE
Bug in "ARP process" section

### ISSUE THREAD

**Issue by SunnyKi** [alex/what-happens-when#159]
Bug in this sentence:
"Now that the network library has the IP address of either our DNS server or the default gateway it can resume its DNS process"

Not "IP address", but "MAC address"

**Comment by Amar-Chaudhari** [comment]
> Not "IP address", but "MAC address"  

Nope, library requires IP Address to send DNS packets. MAC address is needed by the L2 frame to send those packets to that IP address ( or default gateway ).

**Comment by Litee** [comment]
I tend to agree with first comment - description is misleading and wording can be improved. Something like "Now that network library has MAC address ... it can send DNS request and request will be delivered to DNS server or the default gateway".

**Comment by mallika-b** [comment]
Exactly what I wanted to report, as well. 
>The ARP cache is first checked for an ARP entry for our target IP.
The rest of the explanation is about what happens if we don't find the information in cache - so we can't finish the explanation with "Now that the network library has the IP address of either our DNS server or the default gateway" - because we *started* with this IP address!! We didn't have the target MAC (broadcast address was used)

"The Address Resolution Protocol (ARP) feature performs a required function in IP routing. ARP finds the hardware address, also known as Media Access Control (MAC) address, of a host from its known IP address. ARP maintains a cache (table) in which MAC addresses are mapped to IP addresses"
https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/ipaddr_arp/configuration/15-s/arp-15-s-book/Configuring-Address-Resolution-Protocol.html#:~:text=The%20Address%20Resolution%20Protocol%20(ARP,are%20mapped%20to%20IP%20addresses.

**Comment by mallika-b** [comment]
hmm, was considering doing a pull request but looks like it's already been done last Dec:
https://github.com/alex/what-happens-when/pull/304
Unfortunately, hasn't been merged yet, it's Aug now.
